### PR TITLE
Add CurrentProvider Type Definition in index.d.ts

### PR DESCRIPTION
**Pull request by godtaehee**
## Proposed changes

- Describe your changes to communicate to the maintainers why we should accept this pull request.
- If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [x] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/caver-js/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/caver-js)
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- Please leave the issue numbers or links related to this PR here.

## Further comments
I wrote all of my issue in [this issue](https://github.com/klaytn/caver-js/issues/709)

and this is my repository which prove my inconvenient about caver.js

Thanks.

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by jimni1222** [comment]
@godtaehee Thank you for contribution. Please sign CLA. https://github.com/klaytn/caver-js/pull/710#issuecomment-1285551901

File: types/index.d.ts [hunk 1]
```diff
@@ -68,6 +68,7 @@ export class AbstractCaver {
     constructor(provider?: RequestProvider, net?: net.Socket)
 
     version: string
+    currentProvider: RequestProvider
```
Comment (jimni1222) [inline types/index.d.ts:71]: How about setting this type as `provider` which can be imported from `'./packages/caver-core-requestmanager/src'` like below?

```
import { provider, HttpProvider, IpcProvider, WebsocketProvider } from './packages/caver-core-requestmanager/src'
```

**Comment by kjhman21** [comment]
@godtaehee Could you please also add a simple test case to check whether this is accessible from the client code?

**Comment by godtaehee** [comment]
> @godtaehee Thank you for contribution. Please sign CLA. [#710 (comment)](https://github.com/klaytn/caver-js/pull/710#issuecomment-1285551901)

Thanks, I just signed to CLA

File: types/index.d.ts (hunk 1 above)
Comment (godtaehee) [reply types/index.d.ts:71]: oh, I missed the fact which currentProvider type is `not string`. Thanks, That's good catch!!

I modify it.

**Review by jimni1222 — APPROVED** [review]
LGTM! Thank you for contribution!!
After resolving @kjhman21 comment, we can merge this and release :)

**Comment by godtaehee** [comment]
> @godtaehee Could you please also add a simple test case to check whether this is accessible from the client code?

Sure, sir. I will make some test case, and push it!

**Comment by jimni1222** [comment]
> > @godtaehee Thank you for contribution. Please sign CLA. [#710 (comment)](https://github.com/klaytn/caver-js/pull/710#issuecomment-1285551901)
> 
> Thanks, I just signed to CLA
> 
> I have read the CLA Document and I hereby sign the CLA

I think you need to write `I have read the CLA Document and I hereby sign the CLA` comment separate ;-; Still https://github.com/klaytn/caver-js/actions/runs/3293913833/jobs/5430893789 says failed to check

**Comment by godtaehee** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by godtaehee** [comment]
@kjhman21 I just made test code in [`types/test/caver-test.ts` ](https://github.com/klaytn/caver-js/blob/dev/types/test/caver-test.ts)

I can use currentProvider when I use it in js environment using require like [my repository code](https://github.com/godtaehee/caver-js-error-report/blob/main/currentProviderIsExistedinCaverInstance.js)

So I think there is no need for testing in js file which is located in [`root/test/[all of test file].js`](https://github.com/klaytn/caver-js/tree/dev/test)

I searched all of test code in repository. but I can't find the location where I test it.

So If this my [`changes`](https://github.com/klaytn/caver-js/pull/710/files#diff-baead50848d8537b67bb87b9bf6b33c4021ebd63bfa3bc3fccad2575743ace65R37-R39) is not sufficient for testing my PR, Please give me Advice!

Sorry for not well in contributing.

**Review by kjhman21 — APPROVED** [review]
LGTM. Thanks!

**Comment by godtaehee** [comment]
Thanks :D

**Pull request merged**